ebos/equil/initstateequil.hh

```diff
@@ -1721,10 +1730,30 @@ private:
         this->temperature_ = eclState.fieldProps().get_double("TEMPI");
     }
 
+    template <class RMap>
+    void updateInitialSaltConcentration_(const Opm::EclipseState& eclState, const RMap& reg, const Grid& grid)
+    {
+        const auto& tables = eclState.getTableManager();
+        const Opm::TableContainer& saltvdTables = tables.getSaltvdTables();
+        for (size_t i = 0; i < saltvdTables.size(); ++i) {
+            const Opm::SaltvdTable& saltvdTable = saltvdTables.getTable<Opm::SaltvdTable>(i);
+            using TabulatedFunction = typename Opm::Tabulated1DFunction<double>;
+            static TabulatedFunction saltvdTableFunction;
```

Comment:
Just out of curiosity, why is this marked as `static`?  Could you not create the interpolant outside of the outer loop to get the same effect (constructing the interpolant once)?

On another note, I recommend looping the [`activeRegions()`](https://github.com/OPM/opm-grid/blob/fb70c95d5cfc8b1ea62f2f81195557c143a17e47/opm/grid/utility/RegionMapping.hpp#L88) instead of all regions here, e.g.,
```C++
template <class RMap>
void updateInitialSaltConcentration_(const Opm::EclipseState& eclState, const RMap& reg, const Grid& grid)
{
    const auto& tables = eclState.getTableManager();
    const auto& saltvcTables = tables.getSaltvdTables();

    auto saltvdTableFunction = Opm::Tabulated1DFunction<double>{};
    for (const auto& regID : reg.activeRegions()) {
        const auto& saltvdTable = saltvdTables.getTable<Opm::SaltvdTable>(regID);
        saltvdTableFunction.setXYContainers(saltvdTable.getDepthColumn(), saltvdTable.getSaltColumn());

        for (const auto& cell : reg.cells(regID)) {
            const auto depth = UgGridHelpers::cellCenterDepth(grid, cell);
            this->saltConcentration_[cell] = saltvdTableFunction.eval(depth);
        }
    }
}
```